src/flask/routes/oauth2.py

```diff
@@ -0,0 +1,51 @@
+from flask import Blueprint, request, make_response
+
+import jwt
+
+from src.constants.http_status_code import HttpStatus
+from src.env_config import get_env_config_instance
+from src.exceptions.exceptions import UnAuthorizedEndpoint
+from src.lib.repositories.impl_v2.oauth2_repository_impl import Oauth2RepositoryImpl
+from src.utils.oauth2_util import validate_roles_and_scopes
+
+
+def setup_oauth2_routes(ioc):
+
+    oauth2_blueprint = Blueprint("oauth2_blueprint", __name__)
+    oauth2_repository = ioc.get(Oauth2RepositoryImpl)
+
+    @oauth2_blueprint.before_app_request
+    def before_request_middleware():
```

Comment:
What happens if the endpoint is not protected? I think you need to check first if the endpoint is protected and then perform necessary validations if it is protected

Reply:
done. i check for endpoint protect inside function `validate_roles_and_scopes`.

Reply:
done.